.circleci/config.yml

```diff
@@ -264,6 +270,18 @@ jobs:
             - frontend/src/autogen/proto.js
             - lib/streamlit/proto
 
+  # The following jobs inherit from python-3.7.4. In a few cases, there are
+  # steps that are skipped based on the name of the current job (see, e.g.,
+  # "Run frontend tests").
+  python-3.8:
+    <<: *job-template
+    docker:
+      - image: circleci/python:3.8.1
```

Comment:
So we're using 3.8.1 as system python but also installing a venv on top? Where is the version for the venv specified? 

So we'll update the base docker image version but keep the job name as major minor? What are we saving by keeping the job name to major minor if we have to update the base docker image?

Reply:
The venv on top is an existing practice in this CircleCI definition, and I believe it'll use the Docker image's Python version. I believe it's done so that we can utilized CircleCI's caching to capture the `venv/` directory and have all of our installed dependencies captured and restored as part of that cache.

Reply:
Oh, and regarding the job name, right now our `develop` branch declares that certain test suites are required to pass in order to perform merges. Those names are annoying to change, 1) because only Thiago can do it, and 2) because when we want to require a new suite we need everyone to rebase to get the new CircleCI configuration in order to run, and thus pass those named test suites.

Because of this, and because Python micro versions come out regularly, I thought it would be better to keep the names as the overall thing we're trying to test, and leave the micro version we're currently using as an implementation detail that doesn't require us to go through that name-change dance.